bitnami/deepspeed/templates/worker/worker-service.yaml

```
@@ -0,0 +1,81 @@
+  labels: {{- include "common.labels.standard" $ | nindent 4 }}
+    app.kubernetes.io/part-of: deepspeed
+    app.kubernetes.io/component: worker
+    {{- if $root.Values.commonLabels }}
+    {{- include "common.tplvalues.render" (dict "value" $root.Values.commonLabels "context" $) | nindent 4 }}
+    {{- end }}
+    {{- if $root.Values.worker.externalAccess.service.labels }}
+    {{- include "common.tplvalues.render" (dict "value" $root.Values.worker.externalAccess.service.labels "context" $) | nindent 4 }}
+    {{- end }}
+  {{- if or $root.Values.worker.externalAccess.service.annotations $root.Values.commonAnnotations $root.Values.worker.externalAccess.service.loadBalancerAnnotations }}
+  annotations:
+    {{- if and (not (empty $root.Values.worker.externalAccess.service.loadBalancerAnnotations)) (eq (len $root.Values.worker.externalAccess.service.loadBalancerAnnotations) $replicaCount) }}
+    {{ include "common.tplvalues.render" (dict "value" (index $root.Values.worker.externalAccess.service.loadBalancerAnnotations $i) "context" $) | nindent 4 }}
+    {{- end }}
+    {{- if $root.Values.worker.externalAccess.service.annotations }}
+    {{- include "common.tplvalues.render" (dict "value" $root.Values.worker.externalAccess.service.annotations "context" $) | nindent 4 }}
+    {{- end }}
+    {{- if $root.Values.commonAnnotations }}
+    {{- include "common.tplvalues.render" (dict "value" $root.Values.commonAnnotations "context" $) | nindent 4 }}
+    {{- end }}
+  {{- end }}
+spec:
+  type: {{ $root.Values.worker.externalAccess.service.type }}
+  {{- if eq $root.Values.worker.externalAccess.service.type "LoadBalancer" }}
+  {{- if and (not (empty $root.Values.worker.externalAccess.service.loadBalancerIPs)) (eq (len $root.Values.worker.externalAccess.service.loadBalancerIPs) $replicaCount) }}
+  loadBalancerIP: {{ index $root.Values.worker.externalAccess.service.loadBalancerIPs $i }}
+  {{- end }}
+  {{- if $root.Values.worker.externalAccess.service.loadBalancerSourceRanges }}
+  loadBalancerSourceRanges: {{- toYaml $root.Values.worker.externalAccess.service.loadBalancerSourceRanges | nindent 4 }}
+  {{- end }}
+  {{- end }}
+  {{- if or (eq $root.Values.worker.externalAccess.service.type "LoadBalancer") (eq $root.Values.worker.externalAccess.service.type "NodePort") }}
+  externalTrafficPolicy: {{ $root.Values.worker.externalAccess.service.externalTrafficPolicy | quote }}
+  {{- end }}
+  publishNotReadyAddresses: {{ $root.Values.worker.externalAccess.service.publishNotReadyAddresses }}
+  {{- if $root.Values.worker.externalAccess.service.sessionAffinity }}
+  sessionAffinity: {{ $root.Values.worker.externalAccess.service.sessionAffinity }}
+  {{- end }}
+  {{- if $root.Values.worker.externalAccess.service.sessionAffinityConfig }}
+  sessionAffinityConfig: {{- include "common.tplvalues.render" (dict "value" $root.Values.externalAccess.service.sessionAffinityConfig "context" $root) | nindent 4 }}
```

My bad, I was missing the `worker` on my previous suggestion
```suggestion
  sessionAffinityConfig: {{- include "common.tplvalues.render" (dict "value" $root.Values.worker.externalAccess.service.sessionAffinityConfig "context" $root) | nindent 4 }}
```